src/main/java/frc/trigon/robot/subsystems/roller/Roller.java

```diff
@@ -88,31 +92,31 @@ public Command getFullStopCommand() {
     }
 
     private void openRoller() {
-        angleMotor.set(ControlMode.PercentOutput, RollerConstants.OPEN_POWER);
+        rollerIO.setAngleMotorPower(RollerConstants.OPEN_POWER);
     }
 
     private void closeRoller() {
-        angleMotor.set(ControlMode.PercentOutput, RollerConstants.CLOSE_POWER);
+        rollerIO.setAngleMotorPower(RollerConstants.CLOSE_POWER);
     }
 
     private void collect() {
-        collectionMotor.set(RollerConstants.COLLECTION_MOTOR_SPEED);
+        rollerIO.setCollectionMotorPower(RollerConstants.COLLECTION_MOTOR_SPEED);
     }
 
     private void stopCollection() {
-        collectionMotor.stopMotor();
+        rollerIO.stopCollectionMotor();
     }
 
     private void stopAngleMotor() {
-        angleMotor.set(ControlMode.Disabled, 0);
+        rollerIO.stopAngleMotor();
     }
 
     private boolean isOpen() {
-        return !RollerConstants.FORWARD_LIMIT_SWITCH.get();
+        return !ToohardRollerConstants.FORWARD_LIMIT_SWITCH.get();
     }
 
     private boolean isClosed() {
-        return !RollerConstants.BACKWARD_LIMIT_SWITCH.get();
+        return !ToohardRollerConstants.BACKWARD_LIMIT_SWITCH.get();
     }
```

Comment:
You can't have hardware information being read directly. This should go through the inputs class.
Also, never mention io constants in the main subsystem class